### PR TITLE
Fix Arch Linux build

### LINKED ISSUE #12: No rule to make target 'build/SDL.lo'
Hi,

I have been trying to compile Oni2 in ArchLinux but I keep getting the following error:

`make: *** No rule to make target 'build/SDL.lo', needed by 'build/libSDL2.la'.  Stop.`

[Here](https://pastebin.com/veSf5UDT) is the output of `esy build` which includes the output of ./configure and make.

I have everything listed [here](https://github.com/revery-ui/revery/wiki/Building-&-Installing) installed and I'm suspecting it is not a missing dependency because I'm able to compile it using cmake (the official Arch package is also built using [cmake](https://git.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/sdl2))

I'm pretty sure I'll figure it out eventually but I appreciate if anyone has any leads

### PR DESCRIPTION
**Pull request by taylon**
Hi,

I finally found some time to try and fix #12 which is an issue that breaks the build in ArchLinux. 

Many folks using ArchLinux will enjoy using Vim, so I think that having Oni2 building in ArchLinux might be important for the project :)

I didn't have time to try to setup an Arch build in azure pipelines, but I can try that next week

### CODE REVIEW

**Comment by bryphe** [comment]
Hi @taylon , thanks a lot for contributing this fix!

> Many folks using ArchLinux will enjoy using Vim, so I think that having Oni2 building in ArchLinux might be important for the project :)

Definitely 😎 Appreciate you diving to this level to get it working! A couple questions inline, but the overall approach looks good and it builds everywhere.

File: build-scripts/esybuild.sh [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  cmake "$SOURCE_PATH" \
+    -DCMAKE_INSTALL_PREFIX="$INSTALL_PREFIX" \
+    -DSDL_STATIC=OFF \
+    -DSDL_DLOPEN=ON \
+    -DARTS=OFF \
+    -DESD=OFF \
+    -DNAS=OFF \
+    -DALSA=ON \
+    -DPULSEAUDIO_SHARED=ON \
+    -DVIDEO_WAYLAND=OFF \
```
Comment (bryphe) [inline build-scripts/esybuild.sh]: Is there a reason we need to disable `WAYLAND` here? I have a couple experimenting with setting `SDL_VIDEODRIVER=wayland`, would turning this off preclude that?

File: build-scripts/esybuild.sh [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    -DRPATH=OFF \
+    -DCLOCK_GETTIME=ON \
+    -DJACK_SHARED=ON
+}
+
+setup_linux_generic() {
```
Comment (bryphe) [inline build-scripts/esybuild.sh]: Moving all of the linux build to CMake seems reasonable to me. Thanks for preserving the existing behavior, though!

File: package.json [hunk 3]
```diff
@@ -5,27 +5,27 @@
...
     "build": [
-        ["bash", "-c", "#{os == 'windows' ? 'SHELL=bash CC=x86_64-w64-mingw32-gcc WINDRES=x86_64-w64-mingw32-windres SDL_THREAD_PTHREAD=1 ./configure --prefix=$cur__install --enable-pthreads' : 'echo'}"],
-        ["bash", "-c", "#{os == 'darwin' ? './configure --enable-shared=no --enable-static=yes --disable-video-x11 --prefix=$cur__install' : 'echo'}"],
-        ["bash", "-c", "#{os == 'linux' ? './configure --prefix=$cur__install' : 'echo'}"],
-        ["make"],
-        ["make", "install"]
+      [
+        "bash",
+        "-c",
+        "./build-scripts/esybuild.sh --os=#{os}"
```
Comment (bryphe) [inline package.json:11]: Thanks for cleaning up these inline scripts and making them more readable 👍

**Comment by taylon** [comment]
Alright, so I investigated further and noticed that other people were having issues in Arch for other reasons too like https://github.com/onivim/oni2/issues/1778

So I figured that the initial issue was actually because I was running `dash` in my system by default and that was causing issues when generating the Makefile, so I added `CONFIG_SHELL=bash` to prevent that issue.

Then the other issue reported is because of gcc 10 (which is the default in Arch) which sets the -fno-common option by default and that makes the build fail as well...

So with both fixes we can build on Arch and Ubuntu just fine :)

**Comment by bryphe** [comment]
> So with both fixes we can build on Arch and Ubuntu just fine :)

Awesome! Thanks for your work on this @taylon  - looks great 👍 

Appreciate the details too on both the issues hit (the shell issue, and gcc 10).

**Comment by arthur-mts** [comment]
Thanks for thinking in Arch <3

**Pull request merged**